File: ios/CouchbaseLite.m
#import <React/RCTBridgeModule.h>
#import <React/RCTConvert.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(CouchbaseLite, RCTEventEmitter)

RCT_EXTERN_METHOD(
                  run: (NSDictionary *)obj
                  resolver: (RCTPromiseResolveBlock)resolve
                  rejecter: (RCTPromiseRejectBlock)reject
                  )

RCT_EXTERN_METHOD(supportedEvents)

@end

File: ios/CouchbaseLite.swift
@objc(CouchbaseLite)
class CouchbaseLite: RCTEventEmitter {
  @objc
  func run(
    _ obj: NSDictionary, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock
  ) {
    let dict = SafeDictionary(dict: RCTConvert.nsDictionary(obj))

    RNTask.run(dict, self)
      .fold({ reject("", $0, nil) }, { resolve($0) })
  }
  
  override func supportedEvents() -> [String] {
    return [RNReplicator.StatusEventName]
  }
}
